yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+    /// @param amtInWei The amount (in wei/1e-18 ETH) to convert to want tokens.
+    /// @return The amount in want tokens.
+    function ethToWant(uint256 amtInWei)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        // TODO: Create an accurate price oracle.
+        return amtInWei;
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is
+    ///         meant to perform any adjustments to the core position(s) of this
+    ///         strategy, given what change the vault made in the investable
+    ///         capital available to the strategy. All free capital in the
+    ///         strategy after the report was made is available for reinvestment.
+    ///         This strategy implements the aforementioned behavior by taking
+    ///         its balance of the vault's underlying token and depositing it to
+    ///         the tBTC v2 Saddle reward pool.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    function adjustPosition(uint256 debtOutstanding) internal override {
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance > 0) {
+            want.safeIncreaseAllowance(lpRewardsTbtcV2Saddle, wantBalance);
+            ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).stake(wantBalance);
+        }
+    }
+
+    /// @notice Withdraws a portion of the vault's underlying token from
+    ///         the Saddle reward pool.
+    /// @param amount Amount to withdraw.
+    /// @return Amount withdrawn.
+    function withdrawSome(uint256 amount) internal returns (uint256) {
+        amount = Math.min(amount, balanceOfPool());
+        uint256 initialWantBalance = balanceOfWant();
+        // Withdraw some vault's underlying tokens but do not claim the rewards
+        // accumulated so far.
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).withdraw(amount);
+        return balanceOfWant().sub(initialWantBalance);
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to liquidate up to amountNeeded of want token of this strategy's
+    ///         positions, irregardless of slippage. Any excess will be
+    ///         re-invested with adjustPosition. This function should return
+    ///         the amount of want tokens made available by the liquidation.
+    ///         If there is a difference between them, loss indicates whether
+    ///         the difference is due to a realized loss, or if there is some
+    ///         other situation at play (e.g. locked funds). This function is
+    ///         used during emergency exit instead of prepareReturn to
+    ///         liquidate all of the strategy's positions back to the vault.
```

Comment:
I think the part 
```
This function is used during emergency exit instead of prepareReturn to liquidate all of the strategy's positions back to the vault.
```
should refer to `liquidateAllPositions` (see `BaseStrategy.sol` and its `@dev` doc section in this contract) so we can remove this part from here. This mistake was made in existing strategies because we switched Yearn version during the development. The older version didn't contain `liquidateAllPositions` yet and its work was done by `liquidatePosition`. Can we also change that for `CurveVoterProxyStrategy` and `ConvexStrategy` by the way? 🙏

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/13/commits/5f2ce7710c6f13266ff93f1403dab0131da10e68.

Reply:
Perfect! Thanks.